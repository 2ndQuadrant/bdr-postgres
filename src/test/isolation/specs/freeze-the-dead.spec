# Test for interactions of tuple freezing with dead, as well as recently-dead
# tuples using multixacts via FOR KEY SHARE.
setup
{
  CREATE TABLE tab_freeze (
    id int PRIMARY KEY,
    name char(3),
    x int);
  INSERT INTO tab_freeze VALUES (1, '111', 0);
  INSERT INTO tab_freeze VALUES (3, '333', 0);
}

teardown
{
  DROP TABLE tab_freeze;
}

session "s1"
setup				{ BEGIN; }
step "s1_update"	{ UPDATE tab_freeze SET x = x + 1 WHERE id = 3; }
step "s1_commit"	{ COMMIT; }
step "s1_vacuum"	{ VACUUM FREEZE tab_freeze; }

session "s2"
setup				{ BEGIN; }
step "s2_key_share"	{ SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE; }
step "s2_commit"	{ COMMIT; }
