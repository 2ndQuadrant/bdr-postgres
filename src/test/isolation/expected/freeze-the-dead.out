Parsed test spec with 2 sessions

starting permutation: s1_update s1_commit s1_vacuum s2_key_share s2_commit
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s2_commit: COMMIT;

starting permutation: s1_update s1_commit s2_key_share s1_vacuum s2_commit
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_vacuum: VACUUM FREEZE tab_freeze;
step s2_commit: COMMIT;

starting permutation: s1_update s1_commit s2_key_share s2_commit s1_vacuum
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s2_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;

starting permutation: s1_update s2_key_share s1_commit s1_vacuum s2_commit
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;
step s2_commit: COMMIT;

starting permutation: s1_update s2_key_share s1_commit s2_commit s1_vacuum
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_commit: COMMIT;
step s2_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;

starting permutation: s1_update s2_key_share s2_commit s1_commit s1_vacuum
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s2_commit: COMMIT;
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;

starting permutation: s2_key_share s1_update s1_commit s1_vacuum s2_commit
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;
step s2_commit: COMMIT;

starting permutation: s2_key_share s1_update s1_commit s2_commit s1_vacuum
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s2_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;

starting permutation: s2_key_share s1_update s2_commit s1_commit s1_vacuum
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s2_commit: COMMIT;
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;

starting permutation: s2_key_share s2_commit s1_update s1_commit s1_vacuum
step s2_key_share: SELECT id FROM tab_freeze WHERE id = 3 FOR KEY SHARE;
id             

3              
step s2_commit: COMMIT;
step s1_update: UPDATE tab_freeze SET x = x + 1 WHERE id = 3;
step s1_commit: COMMIT;
step s1_vacuum: VACUUM FREEZE tab_freeze;
